apps/theming-designer/src/components/FabricSlotWidget.tsx

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { IThemeRules, FabricSlots, IThemeSlotRule } from 'office-ui-fabric-react/lib/ThemeGenerator';
+import { IColor } from 'office-ui-fabric-react/lib/Color';
+import { Stack } from 'office-ui-fabric-react/lib/Stack';
+import { mergeStyles } from '@uifabric/merge-styles';
+import { ColorPicker } from 'office-ui-fabric-react/lib/ColorPicker';
+import { Callout, DirectionalHint } from 'office-ui-fabric-react/lib/Callout';
+import { Text } from 'office-ui-fabric-react';
+
+export interface IFabricSlotWidgetProps {
+  slot: FabricSlots;
+  onFabricPaletteColorChange: (newColor: IColor | undefined, fabricSlot: FabricSlots) => void;
+  themeRules?: IThemeRules;
+  directionalHint?: DirectionalHint;
+}
+
+export interface IFabricSlotWidgetState {
+  isColorPickerVisible: boolean;
+  colorPickerElement: HTMLElement | null;
+}
+
+const slotClassName = mergeStyles({
+  display: 'flex',
+  alignItems: 'center',
+  overflow: 'auto'
+});
+
+const fabricPaletteColorBox = mergeStyles({
+  width: 15,
+  height: 15,
+  display: 'inline-block',
+  left: 5,
+  top: 5,
+  border: '1px solid black',
+  flexShrink: 0
+});
+
+export class FabricSlotWidget extends React.Component<IFabricSlotWidgetProps, IFabricSlotWidgetState> {
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      isColorPickerVisible: false,
+      colorPickerElement: null
+    };
+
+    this._onColorBoxClick = this._onColorBoxClick.bind(this);
+    this._onCalloutDismiss = this._onCalloutDismiss.bind(this);
+  }
+
+  public render() {
+    const { isColorPickerVisible, colorPickerElement } = this.state;
+    const slotRule = this.props.themeRules![FabricSlots[this.props.slot]];
+    return (
+      <div key={slotRule.name} className={slotClassName}>
+        <Stack horizontal gap={5}>
+          <div
+            key={slotRule.name}
+            className={fabricPaletteColorBox}
+            style={{ backgroundColor: slotRule.color!.str }}
+            onClick={this._onColorBoxClick}
+          />
+          {isColorPickerVisible && (
+            <div>
+              <Callout
```

Comment:
Do you have to put the `Callout` inside a `div` here? Couldn't it be on its own?

This could maybe affect layout if removing so I'm genuinely asking here.

Reply:
I don't think so actually - removed it in ThemeDesignerColorPicker.tsx also!